magefiles/rulesengine/repos/integration_service.go

```diff
@@ -0,0 +1,68 @@
+package repos
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/konflux-ci/e2e-tests/magefiles/rulesengine"
+	"k8s.io/klog"
+)
+
+var IntegrationServiceCICatalog = rulesengine.RuleCatalog{IntegrationServiceCIRule}
+
+var IntegrationServiceCIRule = rulesengine.Rule{Name: "Integration-service repo CI Workflow Rule",
+	Description: "Execute the full workflow for e2e-tests repo in CI",
+	Condition: rulesengine.All{
+		&IntegrationServiceRepoSetDefaultSettingsRule,
+		rulesengine.Any{&InfraDeploymentsPRPairingRule, rulesengine.None{&InfraDeploymentsPRPairingRule}},
+		&PreflightInstallGinkgoRule,
+		&BootstrapClusterWithSprayProxyRuleChain,
+		&IntegrationServiceRepoRunTestsRule,
+	},
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(ExecuteTestAction)},
+}
+
+var BootstrapClusterWithSprayProxyRuleChain = rulesengine.Rule{Name: "BoostrapCluster with SprayProxy RuleChain",
```

Comment:
Do we need this here. This Boostrap Chain looks like it can live in the common.go?

Reply:
yes I think so, thanks. I will update this once https://github.com/konflux-ci/e2e-tests/pull/1344 is merged